go/cmd/vtgate/vtgate.go

```diff
@@ -68,8 +127,23 @@ func main() {
 				log.Errorf("unknown tablet type: %v", ttStr)
 				continue
 			}
-			tabletTypes = append(tabletTypes, tt)
+			//same restriction logic as TabletServer's IsServingType()
```

Comment:
This comment is good. If you don't mind spending a bit more time, it might be worth extracting `IsServingType()` into a function and using it here.

Reply:
good suggestion, will do

Reply:
@deepthi   Done